crates/bevy_ecs/src/system/system.rs

```diff
@@ -66,14 +66,20 @@ pub trait System: Send + Sync + 'static {
     ///
     /// For [read-only](ReadOnlySystem) systems, see [`run_readonly`], which can be called using `&World`.
     ///
+    /// Unlike [`System::run_unsafe`], this will apply deferred parameters *immediately*.
+    ///
     /// [`run_readonly`]: ReadOnlySystem::run_readonly
     fn run(&mut self, input: Self::In, world: &mut World) -> Self::Out {
-        let world = world.as_unsafe_world_cell();
-        self.update_archetype_component_access(world);
-        // SAFETY:
-        // - We have exclusive access to the entire world.
-        // - `update_archetype_component_access` has been called.
-        unsafe { self.run_unsafe(input, world) }
+        let ret = {
+            let world = world.as_unsafe_world_cell();
+            self.update_archetype_component_access(world);
+            // SAFETY:
+            // - We have exclusive access to the entire world.
+            // - `update_archetype_component_access` has been called.
+            unsafe { self.run_unsafe(input, world) }
+        };
```

Comment:
Is this block necessary?

Reply:
`apply_deferred` requires a `&mut World`, and these require `UnsafeWorldCell`. The block might be avoidable if we don't shadow the variable name, but I'm still not sure about the scoping for that.

Reply:
I'd prefer we just not shadow it. The scoping should be fine, unsafe-world-cell is usually well-behaved as a reference type.